src/SAFE.Client/SAFE.fs

```diff
@@ -56,71 +58,96 @@ type ApiCall<'TStart, 'TFinished> =
         match this with
         | NotStarted
-        | Loading -> false
+        | Loading _ -> false
         | Loaded _ -> true
 
-    /// Returns whether the `RemoteData<'T>` value is loading or not.
+    /// Returns whether the `RemoteData<'T>` value has been loaded.
+    member this.HasData =
+        match this with
+        | NotStarted
+        | Loading None -> false
+        | Loading(Some _)
+        | Loaded _ -> true
+
+    /// Returns whether the `RemoteData<'T>` value is loading. This will return true for both first-time and refresh-style loads.
     member this.IsStillLoading =
         match this with
-        | Loading -> true
+        | Loading _ -> true
         | NotStarted
         | Loaded _ -> false
 
-    /// Returns whether the `RemoteData<'T>` value has started loading or not.
-    member this.HasStarted =
+    /// Returns whether the `RemoteData<'T>` value is refreshing itself i.e. Loading (Some _)
+    member this.IsRefreshing =
         match this with
-        | NotStarted -> false
-        | Loading
-        | Loaded _ -> true
+        | Loading(Some _) -> true
+        | _ -> false
+
+    /// Returns whether the `RemoteData<'T>` value has not yet started.
+    member this.HasNotStarted =
+        match this with
+        | NotStarted -> true
+        | Loading _
+        | Loaded _ -> false
 
     /// Maps the underlying value of the remote data, when it exists, into another shape.
     member this.Map mapper =
         match this with
         | NotStarted -> NotStarted
-        | Loading -> Loading
+        | Loading None -> Loading None
+        | Loading(Some value) -> Loading(Some(mapper value))
         | Loaded value -> Loaded(mapper value)
 
-    /// Verifies that a `RemoteData<'T>` value is loaded, and that the data satisfies a given requirement.
+    /// Verifies that a `RemoteData<'T>` value has some data loaded (may be Loading or Loaded), and that the data satisfies a given requirement.
     member this.Exists predicate =
         match this with
-        | NotStarted -> false
-        | Loading -> false
+        | NotStarted
+        | Loading None -> false
+        | Loading(Some value)
         | Loaded value -> predicate value
 
-    /// Like `map` but instead of mapping just the value into another type in the `Loaded` case, it will transform the value into potentially a different case of the `RemoteData<'T>` type.
+    /// Like `map` but instead of mapping just the value into another type in the `Loading` or `Loaded` case, it will transform the value into potentially a different case of the `RemoteData<'T>` type.
     member this.Bind binder =
         match this with
-        | NotStarted -> NotStarted
-        | Loading -> Loading
+        | Loading(Some value)
         | Loaded value -> binder value
+        | v -> v
 
-    /// Maps `Loaded` to `Some`, everything else to `None`.
+    /// Maps `Loaded` or `Loading Some` to `Some`, everything else to `None`.
     member this.ToOption() =
```

Comment:
It's slightly weird that `ToOption` is a method (the usual inference I make from methods is that there's some non-trivial calculation going on behind the scenes). A plain value `member this.AsOption` might be preferable, with `ToOption` marked as obsolete.

Reply:
I think at this point we can make breaking changes - or as you say, mark as obsolete and then remove in a future version.